twins/mongodb/twins.go

```diff
@@ -0,0 +1,209 @@
+// Copyright (c) Mainflux
+// SPDX-License-Identifier: Apache-2.0
+
+package mongodb
+
+import (
+	"context"
+
+	"github.com/mainflux/mainflux/twins"
+	"go.mongodb.org/mongo-driver/bson"
+	"go.mongodb.org/mongo-driver/mongo"
+	"go.mongodb.org/mongo-driver/mongo/options"
+)
+
+const (
+	maxNameSize            = 1024
+	twinsCollection string = "twins"
+)
+
+type twinRepository struct {
+	db *mongo.Database
+}
+
+var _ twins.TwinRepository = (*twinRepository)(nil)
+
+// NewTwinRepository instantiates a MongoDB implementation of twin
+// repository.
+func NewTwinRepository(db *mongo.Database) twins.TwinRepository {
+	return &twinRepository{
+		db: db,
+	}
+}
+
+// Save persists the twin
+func (tr *twinRepository) Save(ctx context.Context, tw twins.Twin) (string, error) {
+	if len(tw.Name) > maxNameSize {
+		return "", twins.ErrMalformedEntity
+	}
+
+	coll := tr.db.Collection(twinsCollection)
+
+	if _, err := tr.RetrieveByID(ctx, tw.ID); err == nil {
```

Comment:
@darkodraskovic Did you conside a [unique index](https://docs.mongodb.com/manual/core/index-unique/)? To avoid this lookup before save.

Reply:
@nmarcetic as per discussion with @drasko I've decided to simply use a uuid for the id in the case we use some other database. mongo's `_id` is still relevant for states, since they need to be ordered as time series.

Reply:
But the unique index is not related to _id, it can be used for any field (for example your ID generated by UUID), the `_id` is just an example in the doc so you understand 

```
A unique index ensures that the indexed fields do not store duplicate values; i.e. enforces uniqueness for the indexed fields. By default, MongoDB creates a unique index on the _id field during the creation of a collection.
```

My question is, can you consider using a unique index for ID field (generated by UUID) instead of or manually check uniqueness using `RetrieveByID`